src/apis/web3api.py

```diff
@@ -45,35 +45,60 @@ def get_current_block_number(self) -> Optional[int]:
             self.logger.warning(f"Error while fetching block number: {err}")
             return None
 
-    def get_tx_hashes_by_block(
-        self, start_block: int, end_block: int
-    ) -> Optional[list[str]]:
+    def get_filtered_receipts(
+        self, start_block: int, end_block: int, target: str, topics: list[Any]
+    ) -> Optional[list[Any]]:
         """
-        Function filters hashes by contract address, and block ranges
+        Function filters receipts by contract address, and block ranges
         """
         filter_criteria: FilterParams = {
             "fromBlock": int(start_block),
             "toBlock": int(end_block),
-            "address": self.web_3.to_checksum_address(SETTLEMENT_CONTRACT_ADDRESS),
-            "topics": [
-                HexStr(
-                    "0xa07a543ab8a018198e99ca0184c93fe9050a79400a0a723441f84de1d972cc17"
-                )
-            ],
+            "address": self.web_3.to_checksum_address(target),
+            "topics": topics,
         }
-
         try:
             log_receipts = self.web_3.eth.filter(filter_criteria).get_all_entries()
         except ValueError as err:
             self.logger.warning(f"ValueError while fetching hashes: {err}")
             return None
+        return log_receipts
 
+    def get_tx_hashes_by_block(
+        self, start_block: int, end_block: int
+    ) -> Optional[list[str]]:
+        """
+        Function filters hashes by contract address, and block ranges
+        """
+        topics = [
+            HexStr("0xa07a543ab8a018198e99ca0184c93fe9050a79400a0a723441f84de1d972cc17")
+        ]
+        log_receipts = self.get_filtered_receipts(
+            start_block, end_block, SETTLEMENT_CONTRACT_ADDRESS, topics
+        )
+
+        if log_receipts is None:
+            return None
         settlement_hashes_list = list(
             {log_receipt["transactionHash"].hex() for log_receipt in log_receipts}
         )
-
         return settlement_hashes_list
 
+    def get_incoming_eth_transfers_to_contract_within_block_range(
+        self, start_block: int, end_block: int, target: str
```

Comment:
Is the target always the settlement contract? Or is it WETH address or something else?

Reply:
With the fix, the target is intended now to be any Safe address